File: tests/omp/collapse2.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->

  !! Sequence_basic.intro 2 [cFor "k"];
  !! Omp.parallel [cVarDef "jlast"];
  !! Omp.for_ ~clause:[Collapse 2; LastPrivate ["jlast"; "klast"]] [cFor "k"];
  !! Omp.single [cFun "printf"];

)

File: tests/omp/collapse2.cpp
#include <stdio.h>
void test()
{
  int jlast;
  int klast;
  for (int k=1; k<=2; k++)
     for (int j=1; j<=3; j++)
     {
        jlast=j;
        klast=k;
     }
  printf("%d %d\n", klast, jlast);
}
